pkg/report/cyclonedx.go

```diff
@@ -0,0 +1,122 @@
+package report
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	fos "github.com/aquasecurity/fanal/analyzer/os"
+	"github.com/aquasecurity/fanal/types"
+	"golang.org/x/xerrors"
+
+	cdx "github.com/CycloneDX/cyclonedx-go"
+	"github.com/gorilla/schema"
+)
+
+// CycloneDXWriter implements result Writer
+type CycloneDXWriter struct {
+	Output io.Writer
+}
+
+// Write writes the results in CycloneDX format
+func (cw CycloneDXWriter) Write(report Report) error {
+	bom, err := convertToBom(report)
+	if err != nil {
+		return xerrors.Errorf("failed to convert to bom: %w", err)
+	}
+	if err = cdx.NewBOMEncoder(cw.Output, cdx.BOMFileFormatJSON).Encode(bom); err != nil {
+		return xerrors.Errorf("failed to encode bom: %w", err)
+	}
+
+	return nil
+}
+
+func convertToBom(r Report) (*cdx.BOM, error) {
+	bom := cdx.NewBOM()
+	bom.Metadata = &cdx.Metadata{
+		Timestamp: time.Now().Format(time.RFC3339),
+		Tools: &[]cdx.Tool{
+			{
+				Vendor:  "aquasecurity",
+				Name:    "Trivy",
+				Version: "0.17.1",
+			},
+		},
+		Component: &cdx.Component{
+			Type:   TypeToComponent(string(r.ArtifactType)),
+			Name:   r.ArtifactName,
+			BOMRef: r.ArtifactID,
+			// TODO: Support properties for cdx 1.3, add repo tags and digests
```

Comment:
This is now possible as of cyclonedx-go [v0.4.0](https://github.com/CycloneDX/cyclonedx-go/releases/tag/v0.4.0)

Reply:
@nscuro 
Thank you!!! Deep respect. 

I have a few question. 
Does CycloneDX-go have any plans to support Vulnerability?

I want to use this struct.
https://cyclonedx.org/use-cases/#vulnerability-disclosure

Reply:
No plans currently, as the vulnerability extension is only supported for XML. Working with XML (and especially namespacing in XML) in Go is a pain, which is why I didn't bother implementing it yet.

I see you've asked a similar question in the CDX Slack already. I think @stevespringett or @coderpatros may be able to give some insights in how vulnerabilities are supposed to be or will be handled in CycloneDX going forward.

If the proposed solutions don't help your case with trivy, you could consider using XML output instead, and I'll see how I can integrate the vulnerability extension into cyclonedx-go.

Reply:
@masahiro331 Refer to CycloneDX support for [Vulnerability Disclosure](https://cyclonedx.org/use-cases/#vulnerability-disclosure) and [Security Advisories](https://cyclonedx.org/use-cases/#security-advisories). 

Disclosing vulnerabilities in an SBOM is currently only supported in XML. That will be changing in a big way with v1.4 of the spec. So in terms of design, I would plan for the eventual support for JSON, but it will be changing from its current format.

Reply:
@masahiro331 CycloneDX v1.4 including support for vulnerabilities is now available in [cyclonedx-go v0.5.0](https://github.com/CycloneDX/cyclonedx-go/releases/tag/v0.5.0).